src/home/homed-home.c

```diff
@@ -1210,6 +1225,29 @@ static int home_start_work(Home *h, const char *verb, UserRecord *hr, UserRecord
                         return r;
         }
 
+        if (blobs) {
+                blob_fd_arr = newa(int, hashmap_size(blobs));
```

Comment:
this is ultimately under user control, hence better use dynamic memory rather than stack i.e. newa() → new()